Regenerate requirements.txt to pull in new Sphinx version. Fixes #368.

nathandyer opened the pull request:
## Status

Ready for review

## Description of Changes

* Description: 

Because requirements.txt had not been regenerated in a while, it was stuck on version 3.5.3 of Sphinx, which does not work in Python 3.10.

By regenerating the file using `pip-compile --allow-unsafe --generate-hashes --output-file=requirements.txt requirements.in`, it is now pulling in the 5.1.1 version, which does work correctly with newer python versions.

* Fixes Issue#

368


## Testing

Build the docs by cloning this branch, then running:
```
python3 -m venv .venv
source .venv/bin/activate
pip install --require-hashes -r requirements/requirements.txt
make docs
```

while on a system with Python 3.10 (`python --version`)


## Release 

Appears to run correctly on two of my local systems. I am not aware of any adverse effects of this change, but welcome input and feedback from reviewers.

## Checklist (Optional)

- [  ] Doc linting (`make docs-lint`) passed locally
- [  ] Doc link linting (`make docs-linkcheck`) passed
- [ x ] You have previewed (`make docs`) docs at http://localhost:8000

eloquence commented:
(Marking as `in development` until CI passes - please let us know if you would like help troubleshooting the `importlib-metadata` issue.)

nathandyer commented:
This should be ready for review again. I fixed the issue with `importlib-metadata` (upstream dependencies had changed so it wasn't being pulled in automatically), and also corrected an issue with the language not being specified in `config.py`, which throws a warning on the newer Sphinx version.

zenmonkeykstop reviewed (approved):
LGTM!
